dockertest/dockercmd.py

```diff
@@ -334,6 +334,49 @@ def execute(self, stdin=None):
                                          stdin=stdin, close_fds=True)
         return self.cmdresult
 
+    def wait_for_ready(self, timeout=None, timestep=0.2):
+        """
+        Monitor the output of a container (including docker logs, in
+        case stdout is detached), waiting for the string 'READY'. Return
+        as soon as we see it. If we don't, throw a meaningful exception.
+
+        :raises DockerExecError: on timeout.
+        """
+        end_time = time.time() + timeout
+        while time.time() <= end_time:
+            time.sleep(timestep)
+            if 'READY' in self.stdout:
+                return
+            # Also check docker logs
+            cid = self.container_id
+            if cid is not None:
+                logs = DockerCmd(self.subtest, 'logs', [self.container_id])
```

Comment:
Function signature changed to accept a cid or name; this lets us bypass calling `container_id`, but I'm leaving the function in anyway because the cid param is optional.